scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,172 @@
+)
+from statvar_dcid_generator import get_stat_var_dcid
+
+
+def generate_stat_var_map(spec_dict: dict, column_list: list) -> dict:
+    """Wrapper function for generateColMapBase class to generate column map.
+
+  Args:
+    specDict:
+      A dictionary containing specifications for the different properties of the statistical variable.
+    columnList:
+      A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+
+  Returns:
+    A dict mapping each column to their respective stat_var node definitions.
+    Example output: {
+      "Total Civilian population": {
+        "populationType": "Person",
+        "statType": "measuredValue",
+        "measuredProperty": "Count Person"
+        "armedForceStatus": "Civilian"
+      },
+      "<column-name-2>": {}, .....
+    }
+  """
+    col_map_obj = GenerateColMapBase(spec_dict=spec_dict,
+                                     column_list=column_list)
+    return col_map_obj._generate_stat_vars_from_spec()
+
+
+class GenerateColMapBase:
+    """module to generate a column map given a list of columns of the dataset and a JSON Spec
+
+  Attributes:
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+    delimiter: The delimiting string that is used for tokenising the column name
+  """
+
+    def __init__(self, spec_dict={}, column_list=[], delimiter='!!'):
+        """module init"""
+        self.features = spec_dict
+        self.column_list = column_list
+        self.column_map = {}
+        self.delimiter = delimiter
+
+    def _update_stat_var_dict(self, stat_var, part_list):
+        """changes certain properties of the stat_var dict based on column name"""
+        # if Margin of Error columns are present change the statType
+        if 'Margin of Error' in part_list:
+            stat_var['statType'] = 'marginOfError'
+        return stat_var
+
+    def _generate_stat_vars_from_spec(self):
+        """generates stat_var nodes for each column in column list and returns a dictionary of columns with their respective stat_var nodes
+        Example output: {
+      "Total Civilian population": {
+        "populationType": "Person",
+        "statType": "measuredValue",
+        "measuredProperty": "Count Person"
+        "armedForceStatus": "Civilian"
+      },
+      "<column-name-2>": {}, .....
+    }"""
+        # for each column generate the definition of their respective statistical variable node
+        for col in self.column_list:
+            if 'ignoreColumns' in self.features and col not in self.features[
+                    'ignoreColumns']:
+                self.column_map[col] = self._column_to_statvar(col)
+        #TODO: Enable validating the generate columnMap: _valid_col_map():
+        return self.column_map
```

Comment:
can we check of the same statvar is generated for more than one column?

Reply:
Fair point! I think we might end up with a set of validation rules on which we check the generated column map. At the moment, it is not clear how we will be handling the cases where validation fails, hence I safely marked it as TODO. Do you reckon, the validation of the column map be part of this PR or a separate PR at a later time?

Reply:
Let's move all new features to future PR's. Please add a more specific TODO comment. Please [use this format for TODO's](https://google.github.io/styleguide/pyguide.html#312-todo-comments) - note the space after `#`

Reply:
I have updated the #TODO and the also added a list of [validation rules](https://github.com/sharadshriram/data/blob/acs_common/scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py#L386-L394) that can be good to have in the next PR.